### PR TITLE
server: move regionCache to cache directory.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: server/balancer.go [hunk 1]
```diff
@@ -138,15 +139,15 @@ func (l *balanceLeaderScheduler) Schedule(cluster *clusterInfo) Operator {
 type balanceRegionScheduler struct {
 	opt      *scheduleOption
 	rep      *Replication
-	cache    *idCache
+	cache    *cache.IDTTL
```
Comment (ngaut) [inline server/balancer.go]: The name IDTTL is wired, please figure out a better name

Comment (disksing) [reply server/balancer.go]: Do you have any suggestion? @ngaut

Comment (ngaut) [reply server/balancer.go]: CacheWithTTL

Comment (ngaut) [reply server/balancer.go]: CacheWithTTL

Comment (ngaut) [reply server/balancer.go]: CacheWithTTL

Comment (disksing) [reply server/balancer.go]: Can't use name starts with 'Cache', because the package is 'cache', lint tools will complain about it.

Comment (huachaohuang) [reply server/balancer.go]: TTLUint64?

**Pull request merged**